### PR TITLE
Added Support of ScintillaNET Custom Line Numbers

### PR DESCRIPTION
**Pull request by olaws**
Implemented idea #9 to support ScintillaNET Custom Line Numbers.

### CODE REVIEW

File: ScintillaNet_FindReplaceDialog/GoTo/GoToDialog.cs [hunk 1]
```diff
@@ -139,9 +140,10 @@ public Scintilla Scintilla
 
         #region Constructors
 
-        public GoToDialog()
+        public GoToDialog(int startingLineIndex)
```
Comment (VPKSoft) [inline ScintillaNet_FindReplaceDialog/GoTo/GoToDialog.cs:143]: The constructor should be overloaded to keep backwards compatibly in check.

File: ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs [hunk 2]
```diff
@@ -85,10 +95,9 @@ public void UpdateFindAllResults(FindReplace FindReplace, List<ScintillaNET_Find
 
                 if (startLine == endLine)
                 {
-                    string resultsLinePrefix = string.Format(Properties.Resources.FindAllResults_ResultsLinePrefix, startLine + 1);
+	                string resultsLinePrefix = string.Format("Line {0}: ", startLine + _startingLineNumber);
```
Comment (VPKSoft) [inline ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs:98]: Removing this breaks the localization support as this is also localized in German, so either:
* Change the resource files to accept two parameters
* Leave this as it was before
image

File: Demo/Form1.cs [hunk 3]
```diff
@@ -30,6 +32,10 @@ public Form1()
 			incrementalSearcher1.FindReplace = MyFindReplace;
 
 			findAllResultsPanel1.Scintilla = scintilla1;
+			
+			// The default line number is 1, anything else is a custom line number
+			customStartingLineNumber = 8;
```
Comment (VPKSoft) [inline Demo/Form1.cs:37]: The demo app should have an option to enable the custom line numbering. Either one way &rarr; button click or two-way such as:
image

**Review by VPKSoft — CHANGES_REQUESTED** [review]
I added few notes, otherwise looks good 👍

File: Demo/Form1.cs (hunk 3 above)
Comment (olaws) [reply Demo/Form1.cs]: I've added a new commit to enable\disable the custom line numbering, but I'm new to git and GitHub so I'm not sure if I've added correctly.

File: ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs (hunk 2 above)
Comment (olaws) [reply ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs]: I've rolled this back, but again I'm new to git and GitHub so I'm not sure if I've added correctly.

File: ScintillaNet_FindReplaceDialog/GoTo/GoToDialog.cs (hunk 1 above)
Comment (olaws) [reply ScintillaNet_FindReplaceDialog/GoTo/GoToDialog.cs:148]: I've done this, but again I'm new to git and GitHub so I'm not sure if I've added correctly.

Comment (VPKSoft) [reply ScintillaNet_FindReplaceDialog/GoTo/GoToDialog.cs:148]: This looks good now.

File: ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs (hunk 2 above)
Comment (VPKSoft) [reply ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs]: The rollback might have failed? Your fork doesn't contain the reverted version either: https://github.com/olaws/ScintillaNET-FindReplaceDialog/blob/869340f66eba62de7da99c8adaaf4948c996cb9c/ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs#L113. Perhaps you forgot to push the change?

File: Demo/Form1.cs (hunk 3 above)
Comment (VPKSoft) [reply Demo/Form1.cs]: This looks good.

File: ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs (hunk 2 above)
Comment (VPKSoft) [reply ScintillaNet_FindReplaceDialog/FindAllResults/FindAllResultsPanel.cs]: This looks good now also.

File: Demo/Form1.cs [hunk 4]
```diff
@@ -30,6 +34,8 @@ public Form1()
 			incrementalSearcher1.FindReplace = MyFindReplace;
 
 			findAllResultsPanel1.Scintilla = scintilla1;
+
+			InitNumberMargin();
```
Comment (VPKSoft) [inline Demo/Form1.cs:38]: Could this empty margin be initialized / cleared when toggling the custom line numbers?
image

**Review by VPKSoft — APPROVED** [review]
One issue still with the empty margin being initialized in the demo form constructor, otherwise looks good 👍

**Review by VPKSoft — CHANGES_REQUESTED** [review]
One issue still with the empty margin being initialized in the demo form constructor, otherwise looks good 👍

**Review by VPKSoft — APPROVED** [review]
Thanks - looks good.

**Comment by VPKSoft** [comment]
Thanks for the contribution - I released a new version with your changes including the [NuGet](https://www.nuget.org/packages/ScintillaNET_FindReplaceDialog/1.5.5) package.

**Pull request merged**